ecl/hql/hqlutil.cpp

```diff
@@ -5556,36 +5620,44 @@ IHqlExpression * extractCppBodyAttrs(unsigned lenBuffer, const char * buffer)
             {
                 if ((i + 1 + 6 < lenBuffer) && memicmp(buffer+i+1, "option", 6) == 0)
                 {
+                    unsigned valueStart, valueEnd;
                     unsigned start = skipSpace(i+1+6, lenBuffer, buffer);
                     unsigned end = start;
                     while (end < lenBuffer && !iseol((byte)buffer[end]))
                         end++;
-                    end = trimSpace(end, buffer);
-                    if (matchOption(start, lenBuffer, buffer, 4, "pure"))
+                    i = end;
+                    if (matchOption(start, end, buffer, 4, "pure", valueStart, valueEnd))
                         attrs.setown(createComma(attrs.getClear(), createAttribute(pureAtom)));
-                    else if (matchOption(start, lenBuffer, buffer, 4, "once"))
+                    else if (matchOption(start, end, buffer, 4, "once", valueStart, valueEnd))
                         attrs.setown(createComma(attrs.getClear(), createAttribute(onceAtom)));
-                    else if (matchOption(start, lenBuffer, buffer, 6, "action"))
+                    else if (matchOption(start, end, buffer, 6, "action", valueStart, valueEnd))
                         attrs.setown(createComma(attrs.getClear(), createAttribute(actionAtom)));
-                    else if (matchOption(start, lenBuffer, buffer, 6, "source"))
+                    else if (matchOption(start, end, buffer, 6, "source", valueStart, valueEnd))
                     {
-                        stripQuotes(start, end, buffer);
-                        Owned<IValue> restOfLine = createUtf8Value(end-start, buffer+start, makeUtf8Type(UNKNOWN_LENGTH, NULL));
-                        OwnedHqlExpr arg = createConstant(restOfLine.getClear());
-                        attrs.setown(createComma(attrs.getClear(), createAttribute(sourceAtom, arg.getClear())));
+                        if ((valueEnd-valueStart) > 0)
```

Comment:
Seems to me you should be giving some sort of error ("value expected?") if there is no value supplied, rather than silently ignoring? Might be easiest to pass a boolean to the matchOption function to indicate whether a value is expected or not (and give an error "value NOT expected" if one is supplied to the options that don't use one).